Realm/RLMAnalytics.mm

```diff
@@ -194,80 +241,207 @@ static auto RLMSysCtl(int *mib, u_int mibSize, size_t *bufferSize) {
+                } else {
+                    featuresDictionary[@"Primitive_Set"] = @1;
+                }
+                continue;
+            }
+            if (property.dictionary) {
+                if (property.type == RLMPropertyTypeObject) {
+                    featuresDictionary[@"Object_Dictionary"] = @1;
+                } else {
+                    featuresDictionary[@"Primitive_Dictionary"] = @1;
+                }
+                continue;
+            }
+
+            switch (property.type) {
+               case RLMPropertyTypeAny:
+                    featuresDictionary[@"Mixed"] = @1;
+                  break;
+               case RLMPropertyTypeObject:
+                    featuresDictionary[@"Object_Link"] = @1;
+                  break;
+                case RLMPropertyTypeLinkingObjects:
+                    featuresDictionary[@"Backlink"] = @1;
+                   break;
+               default:
+                    break;
+            }
+        }
     }
+    return featuresDictionary;
+}
+
+static NSDictionary *RLMConfigurationMetrics(RLMRealmConfiguration *configuration) {
+    RLMSyncConfiguration *syncConfiguration = configuration.syncConfiguration;
+    bool isSync = syncConfiguration != nil;
+    bool isPBSSync = syncConfiguration.partitionValue != nil;
+    bool isFlexibleSync = (isSync && !isPBSSync);
+    auto resetMode = syncConfiguration.clientResetMode;
+
+    bool isCompactOnLaunch = configuration.shouldCompactOnLaunch != nil;
+    bool migrationBlock = configuration.migrationBlock != nil;
+
+    return @{
+        // Sync
+        @"Sync Enabled": isSync ? @"true" : @"false",
+        @"Flx_Sync": isFlexibleSync ? @1 : @0,
+        @"Pbs_Sync": isPBSSync ? @1 : @0,
+
+        // Client Reset
+        @"CR_Recover_Discard": (isSync && resetMode == RLMClientResetModeRecoverOrDiscardUnsyncedChanges) ? @1 : @0,
+        @"CR_Recover": (isSync && resetMode == RLMClientResetModeRecoverUnsyncedChanges) ? @1 : @0,
+        @"CR_Discard": (isSync && resetMode == RLMClientResetModeDiscardUnsyncedChanges) ? @1 : @0,
+        @"CR_Manual": (isSync && resetMode == RLMClientResetModeManual) ? @1 : @0,
+
+        // Configuration
+        @"Compact_On_Launch": isCompactOnLaunch ? @1 : @0,
+        @"Schema_Migration_Block": migrationBlock ? @1 : @0,
+    };
+}
+
+void RLMSendAnalytics(RLMRealmConfiguration *configuration, RLMSchema *schema) {
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+        if (getenv("REALM_DISABLE_ANALYTICS") || !RLMIsDebuggerAttached() || RLMIsRunningInPlayground()) {
+            return;
+        }
+        NSArray *urlStrings = @[@"https://data.mongodb-api.com/app/realmsdkmetrics-zmhtm/endpoint/metric_webhook/metric?data=%@"];
+
+        NSDictionary *baseMetrics = RLMBaseMetrics();
+        NSDictionary *schemaMetrics = RLMSchemaMetrics(schema);
+        NSDictionary *configurationMetrics = RLMConfigurationMetrics(configuration);
```

Comment:
configuration needs to be copied before the dispatch_async() as it's a mutable object and the user could mutate after opening a Realm while we're reading it on a background thread.